tests/tasks/files/test_operations.py

```diff
@@ -144,3 +144,54 @@ def test_remove_directory(self, tmpdir):
 
         Remove(path=source).run()
         assert not source.exists()
+
+
+class TestGlob:
+    def test_initialization(self):
+        ld = Glob(path="/some/path")
+        assert ld.path == "/some/path"
+        assert ld.pattern == "*"
+
+        ld = Glob()
+        assert ld.path == ""
+
+    def test_path_not_provided(self, tmpdir):
+        ld = Glob()
+        with pytest.raises(ValueError, match="No `path` provided"):
+            ld.run()
+
+    def test_list_dir(self, tmpdir):
+        source = tmpdir.mkdir("source").join("testfile")
+        source.write_binary(b"test")
+
+        ld = Glob(path=Path(tmpdir).joinpath("source"))
+        res = ld.run()
+
+        assert res[0] == Path(str(source))
+        assert isinstance(res[0], Path)
+
+    def test_list_dir_recrusive(self, tmpdir):
+        source = tmpdir.mkdir("source").mkdir("dir1").join("testfile")
+        source.write_binary(b"test")
```

Comment:
I'm not sure about `tmpdir` because it's a weird pytest path object, but if you use the `tmp_path` fixture, https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmp-path-fixture you can do the simpler

```suggestion
        source.touch()
```

It also has `write_text("foo")` which makes more sense than writing old-style binary strings, but I'm also not sure off the top of my head if pytest's path object supports that :)

Reply:
Yep, `tmp_path` worked well and is a lot cleaner!